### DISCUSSION TITLE
How to move to previous file?

### DISCUSSION THREAD

**Discussion by xzbdmw** [zed-industries/zed#6616] [Feature Requests]
If I use cmd+p, I must use down or up to select the nearest file, which is unconvenient, and in Jetbrains have a switcher, which I can click option+tab and hold Option while continue clicking tab to switch with recent files if I want more other than most recent, and more importantly, if I release the option keyboard, the select panel is gone. It is very intuative

**Comment by andrejsm** [comment]
As of v0.107 `cmd-p` cycles through files. Which is great and significantly improves DX.
What is missing is, I dare to say, expected behavior where upon invocation of the command it selects the file next in line, not the one that is already open. I say expected as this is how web browsers do it, operating systems do it, etc.

Either way, thank you Zed team for bringing this small but very impactful command!

**Comment by samjeffress** [comment]
There is now the Tab Switcher which is pretty much what you've described. `ctrl-tab` is the default key binding. I'm not sure what version it came in on but has been around since March 2024 - https://github.com/zed-industries/zed/issues/7653. 

For some reason the keybinding is only in the `Workspace` context - I've added it to the `Editor` too which is where I do most of my file switching. 

The keymap looks like: 
```
{
    "context": "Editor",
    "bindings": {
      "ctrl-tab": "tab_switcher::Toggle",
      "ctrl-shift-tab": ["tab_switcher::Toggle", { "select_last": true }]
    }
  },
```

**Comment by kassi** [comment]
I still find this very unintuitive.
Is it possible to not show the panel with the file names and also use the order from left to right instead of jumping around?
I'm used to have a key binding that just switches the next (right) open tab, and another one for the left one. When I use the keybinding described above, it always opens the file name panel and doesn't switch, then I have to repeat undtil it actually moves - and it doesn't move to the next in line, but some other file.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Reply by MrBns** [reply]
No, discord cannot be a better place. I came here from google search. github discussions are available to google search. and this is the main benefit of having github discussion. 

please keep it open. because people outside discord won't find anything with the help of AI, or google search.

**Comment by medzhidov** [comment]
How to make this work like in browser? option+cmd+left switch to previous tab, option+cmd+right switch to next tab?

**Comment by MrBns** [comment]
Huuuhh.. that was the most weird reason I didn't use zed.  I remember people complained about it very early than you. still zed didn't care. its just a shortcut issue. 

if they don't think this is an issue, people who come from vscode will find it so uncomfortable. becuase `ctrl + p` is the most used, may be the top one used shortcut in vscode.
